jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoSender.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) 2016,2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.jmeter.client;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.jmeter.samplers.SampleResult;
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.client.MessageSender;
+import org.eclipse.hono.client.RegistrationClient;
+import org.eclipse.hono.client.impl.HonoClientImpl;
+import org.eclipse.hono.connection.ConnectionFactory;
+import org.eclipse.hono.connection.ConnectionFactoryImpl;
+import org.eclipse.hono.jmeter.HonoSampler;
+import org.eclipse.hono.jmeter.HonoSenderSampler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonClientOptions;
+
+/**
+ * Sender, which connects to directly to Hono; asynchronous API needs to be used synchronous for JMeters threading model
+ */
+public class HonoSender {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HonoSender.class);
+
+    private ConnectionFactory  connectionFactory;
+    private HonoClient         honoClient;
+    private RegistrationClient registrationClient;
+    private String             token;
+    private MessageSender      messageSender;
+    private Vertx              vertx = Vertx.vertx();
+    private HonoSenderSampler  sampler;
+
+    public HonoSender(final HonoSenderSampler sampler) throws InterruptedException {
+        this.sampler = sampler;
+
+        // hono config
+        connectionFactory = ConnectionFactoryImpl.ConnectionFactoryBuilder.newBuilder()
+                .disableHostnameVerification()
+                .host(sampler.getHost())
+                .name(sampler.getContainer())
+                .user(sampler.getUser())
+                .password(sampler.getPwd())
+                .port(sampler.getPort())
+                .trustStorePath(sampler.getTrustStorePath())
+                .vertx(vertx)
+                .build();
+
+        LOGGER.debug("create hono sender - tenant: " + sampler.getTenant() + "  deviceId: " + sampler.getDeviceId());
```

Comment:
this might be true for this occurrence. for the sake of consistency with other log statements I'd rather have this changed to use the generally preferable way of creating log statements ...